Updated BERTweet model card. (#37981)
huggingface/transformers: docs/source/en/model_doc/bertweet.md

    ">
</div>

## BERTweet

[BERTweet](https://huggingface.co/papers/2005.10200) is basically BERT, but for tweets. It shares the same architecture as BERT-base, but it’s been pre-trained specifically on English Twitter data using the RoBERTa-style training setup. That means it’s way more familiar with tweet lingo — hashtags, abbreviations, usernames, and all the quirks of social media.

> Because of this Twitter-native training, BERTweet performs really well on tweet-related NLP tasks like part-of-speech tagging, named entity recognition, and text classification — even outperforming big models like RoBERTa-base and XLM-R-base in those areas. 

Reviewer: ```suggestion
```

You can find all the original BERTweet checkpoints under the [VinAI Research](https://huggingface.co/vinai?search_models=BERTweet) organization.

> [!TIP]

> > This implementation is the same as BERT, except for tokenization method. Refer to [BERT](https://huggingface.co/docs/transformers/en/model_doc/bert) documentation for API reference information.

Reviewer: ```suggestion
> Refer to the [BERT](./bert) docs for more examples of how to apply BERTweet to different language tasks.
```

The example below demonstrates how to predict the `<mask>` token with [`Pipeline`], [`AutoModel`], and from the command line.

<hfoptions id="usage">
<hfoption id="Pipeline">

```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="fill-mask",
    model="vinai/bertweet-base",
    torch_dtype=torch.float16,
    device=0
)
pipeline("Plants create <mask> through a process known as photosynthesis.")
```
</hfoption>
<hfoption id="AutoModel">

```py
import torch
from transformers import AutoModelForMaskedLM, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained(

>    model_id,

Reviewer: ```suggestion
   "vinai/bertweet-base",
```

)
model = AutoModelForMaskedLM.from_pretrained(
    "vinai/bertweet-base",
    torch_dtype=torch.float16,
    device_map="auto"
)
inputs = tokenizer("Plants create <mask> through a process known as photosynthesis.", return_tensors="pt").to("cuda")

with torch.no_grad():
    outputs = model(**inputs)
    predictions = outputs.logits

masked_index = torch.where(inputs['input_ids'] == tokenizer.mask_token_id)[1]
predicted_token_id = predictions[0, masked_index].argmax(dim=-1)
predicted_token = tokenizer.decode(predicted_token_id)

print(f"The predicted token is: {predicted_token}")
```

</hfoption>
<hfoption id="transformers CLI">

```bash

> echo -e "Plants create <mask> through a process known as photosynthesis." | transformers-cli run --task fill-mask --model vinai/bertweet-base --device 0

Reviewer: ```suggestion
echo -e "Plants create <mask> through a process known as photosynthesis." | transformers run --task fill-mask --model vinai/bertweet-base --device 0
```

Author: Hi @stevhliu, in a previous PR (for BART) you mentioned that I should leave the transformers-cli cmd as is, should I do that here as well? I might have mis understood what you meant.

Reviewer: Yeah sorry, this should still be `transformers-cli run ...`

```

</hfoption>
</hfoptions>

## Notes

> - Use the AutoTokenizer or BertweetTokenizer — it’s preloaded with a custom vocabulary adapted to tweet-specific tokens like hashtags (#), mentions (@), emojis, and common abbreviations.

Reviewer: ```suggestion
- Use the [`AutoTokenizer`] or [`BertweetTokenizer`] because it’s preloaded with a custom vocabulary adapted to tweet-specific tokens like hashtags (#), mentions (@), emojis, and common abbreviations. Make sure to also install the [emoji](https://pypi.org/project/emoji/) library.
```

> - Inputs should be padded on the right (e.g., padding="max_length") to align with the positional encoding behavior.

Reviewer: ```suggestion
- Inputs should be padded on the right (`padding="max_length"`) because BERT uses absolute position embeddings.
```

> - Since it was pre-trained on tweets, it performs best on short, informal, and social media-style text.

Reviewer: ```suggestion
```
